src/slicer_connection_interface.py

```diff
@@ -0,0 +1,110 @@
+import asyncio
+import pyigtl
+import time as t
+import logging
+
+from src.common.api.mct_request_series import MCTRequestSeries
+from src.common.structures.component_role_label import COMPONENT_ROLE_LABEL_DETECTOR, COMPONENT_ROLE_LABEL_POSE_SOLVER
+from src.controller.mct_controller import MCTController
+from src.controller.structures.mct_component_address import MCTComponentAddress
+from ipaddress import IPv4Address
+
+from src.pose_solver.api.add_target_marker_request import AddTargetMarkerRequest
+from src.pose_solver.api.set_reference_marker_request import SetReferenceMarkerRequest
+
+from src.controller.structures.connection import Connection
+
+pose_solver_label = "p"
+detector_labels_and_IPs = [
+    ("d101",IPv4Address('192.168.0.101')),
+    ("d102",IPv4Address('192.168.0.102'))]
+
+server = pyigtl.OpenIGTLinkServer(port=18944,local_server=False)
+
+async def update(controller):
+    try:
+        await controller.update()
+    except Exception as e:
+        controller.add_status_message(
+            severity="error",
+            message=f"Exception occurred in controller loop: {str(e)}")
+        
+    ps_frame = controller.get_live_pose_solver_frame(pose_solver_label)
+    timestamp = ps_frame.timestamp_utc_iso8601
+
+    if len(ps_frame.target_poses) > 0:
+        target_poses = ps_frame.target_poses[0]
+        message = pyigtl.TransformMessage(
+            matrix=target_poses.object_to_reference_matrix.as_numpy_array(),
+            # timestamp=target_poses.solver_timestamp_utc_iso8601,
+            device_name=target_poses.target_id
+        )
+        server.send_message(message,wait=True)
+
+    event_loop: asyncio.AbstractEventLoop = asyncio.get_running_loop()
+    event_loop.create_task(update(controller))
+
+async def main():
+    logging.basicConfig(level=logging.INFO)
+
+    controller = MCTController(
+        serial_identifier="controller",
+        send_status_messages_to_logger=True)
+    
+    controller.add_status_message(
+            severity="info",
+            message=f"Slicer client connected")
+
+    controller.add_connection(MCTComponentAddress(
+            label=pose_solver_label,
+            role=COMPONENT_ROLE_LABEL_POSE_SOLVER,
+            ip_address=IPv4Address("127.0.0.1"),
+            port=8000
+        ))
+    
+    for (label, ip) in detector_labels_and_IPs:
+        controller.add_connection(MCTComponentAddress(
+                label=label,
+                role=COMPONENT_ROLE_LABEL_DETECTOR,
+                ip_address=ip,
+                port=8001
+            ))
+
+    controller.start_up()
+
+    for idx, report in enumerate(controller.get_connection_reports()):
+        while report.status != Connection.State.RUNNING:
+            await controller.update()
+            report = controller.get_connection_reports()[idx]
```

Comment:
If there is a failure then this could run indefinitely. It's probably ok for now (internal use), but it's not ideal. You can try `while controller.is_transitioning()` instead